src/main/scala/org/hammerlab/guacamole/alignment/AffineGapPenaltyAlignment.scala

```diff
@@ -0,0 +1,157 @@
+package org.hammerlab.guacamole.alignment
+
+import breeze.linalg.{ DenseMatrix, argmin }
+import org.hammerlab.guacamole.alignment.AlignmentState.AlignmentState
+
+import scala.annotation.tailrec
+
+object AffineGapPenaltyAlignment {
+
+  /**
+   * Retraces the optimal alignment path from a matrix of scores and transitions
+   *
+   * @param alignmentStates sequenceLength x referenceLength matrix of alignment score at each step
```

Comment:
"alignment score" here is confusing given this parameter, and its data type, are "alignmentStates" and the next parameter is "alignmentScores" / Doubles.